hostinfo/hostname_macos.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build cgo && darwin && !ios
+
+package hostinfo
+
+/*
+#cgo CFLAGS: -x objective-c
+#cgo LDFLAGS: -framework Foundation
+#import <Foundation/Foundation.h>
+
+const char *getHostname() {
+	NSString *hostname = [[NSHost currentHost] localizedName];
```

Comment:
We should not use `NSHost`; it is [deprecated](https://developer.apple.com/documentation/foundation/nshost). We should use [`SCDynamicStoreCopyLocalHostName()`](https://developer.apple.com/documentation/systemconfiguration/1517066-scdynamicstorecopylocalhostname).

Also, `SCDynamicStoreCopyLocalHostName()` is strictly better because it does not include the network name, which saves us a line above.